Set3-Feedback/17/main.cc

```diff
@@ -0,0 +1,21 @@
+//JB: ?
+#include "main.ih"
+
+int main(int const argc, char const *argv[])
+{
+    bool findDouble = false; //JB: COC/MLR
+    for (int argIdx = 1; argIdx != argc; ++argIdx)
+    {
+        string const argument = argv[argIdx];
+        if (argument.find(".") != string::npos)
+        {
+            findDouble = true;
+            break;                              // One double is enough
+        }
+    }
```

Comment:
MLR/COC: Multiple levels of responsibility/constant correctness.

This can be refactored into a helper function:

```cpp
bool findDouble(size_t const argc, char const *argv[]
{
    for (int argIdx = 1; argIdx != argc; ++argIdx)
    {
        string const argument = argv[argIdx];
        if (argument.find(".") != string::npos)
			return true; // one double is enough
	}
	return true;
}
```